types/prosemirror-view/index.d.ts

```diff
@@ -1,100 +1,95 @@
-    handleClick?(view: EditorView, pos: number, event: MouseEvent): boolean
-    handleDoubleClickOn?(view: EditorView, pos: number, node: ProsemirrorNode, nodePos: number, event: MouseEvent, direct: boolean): boolean
-    handleDoubleClick?(view: EditorView, pos: number, event: MouseEvent): boolean
-    handleTripleClickOn?(view: EditorView, pos: number, node: ProsemirrorNode, nodePos: number, event: MouseEvent, direct: boolean): boolean
-    handleTripleClick?(view: EditorView, pos: number, event: MouseEvent): boolean
-    handleContextMenu?(view: EditorView, pos: number, event: MouseEvent): boolean
-    onFocus?(view: EditorView, event: Event): void
-    onBlur?(view: EditorView, event: Event): void
-    domParser?: DOMParser;
-    clipboardParser?: DOMParser;
-    transformPasted?(p: Slice): Slice
-    transformPastedHTML?(p: string): string
-    transformPastedText?(p: string): string
-    nodeViews?: Object;
-    clipboardSerializer?: DOMSerializer;
-    decorations?(p: EditorState): DecorationSet
-    editable?(p: EditorState): boolean
-    attributes?: Object | ((p: EditorState) => Object | void);
-    scrollThreshold?: number;
-    scrollMargin?: number;
-
-  }
-  export interface NodeView {
-    dom?: Node;
-    contentDOM?: Node;
-    update?(node: ProsemirrorNode, decorations: Decoration[]): boolean
-    selectNode?(): void
-    deselectNode?(): void
-    setSelection?(anchor: number, head: number, root: Document): void
-    stopEvent?(event: Event): boolean
-    ignoreMutation?(p: MutationRecord): boolean
-    destroy?(): void
-
-  }
-
-}
\ No newline at end of file
+export class Decoration {
+  spec: AnyObject;
+  static widget(pos: number, dom: Node, spec?: { side?: number | null, stopEvent?: ((event: Event) => boolean) | null, key?: string | null }): Decoration;
+  static inline(from: number, to: number, attrs: DecorationAttrs, spec?: { inclusiveStart?: boolean | null, inclusiveEnd?: boolean | null }): Decoration;
+  static node(from: number, to: number, attrs: DecorationAttrs, spec?: AnyObject): Decoration;
+}
+export interface DecorationAttrs {
+  class?: string | null;
+  style?: string | null;
+  nodeName?: string | null;
+}
+export class DecorationSet {
+  find(start?: number, end?: number): Decoration[];
+  map(mapping: Mapping, doc: Node, options?: { onRemove?: ((decorationSpec: AnyObject) => void) | null }): DecorationSet;
+  add(doc: Node, decorations: Decoration[]): DecorationSet;
+  remove(decorations: Decoration[]): DecorationSet;
+  static create(doc: Node, decorations: Decoration[]): DecorationSet;
+  static empty: DecorationSet;
+}
+export class EditorView {
+  constructor(place: null | Node | ((p: Node) => void) | { mount: Node }, props: EditorProps)
+  state: EditorState;
+  dom: Element;
+  props: EditorProps;
+  update(props: EditorProps): void;
+  setProps(props: EditorProps): void;
```

Comment:
Despite the prosemirror docs saying that props here should be an instance of EditorProps, it's just a simple object, not EditorProps:

> Update the view by updating existing props object with the object given as argument. Equivalent to view.update(Object.assign({}, view.props, props)).